app/api/views/briefs.py

```diff
@@ -1175,12 +1179,22 @@ def download_brief_attachment(brief_id, slug):
     """
     brief = briefs.get(brief_id)
 
+    if not brief:
+        return not_found('File not found')
+
+    all_documents = (
+        brief.data.get('attachments', []) +
+        brief.data.get('requirementsDocument', []) +
+        brief.data.get('responseTemplate', [])
+    )
+
     if (
         hasattr(current_user, 'role') and
         (
             current_user.role == 'buyer' or (
                 current_user.role == 'supplier' and
-                _can_do_brief_response(brief_id, update_only=True)
+                _can_do_brief_response(brief_id, update_only=True) and
+                slug in all_documents
```

Comment:
A few things in this PR have a wider impact than just document editing - this will add a check on document access that ensures the document being requested is in the opportunity's current data (so removed or replaced documents become inaccessible).